File: Tessoku_of_Competitive_Programming/B06 - Lottery.py
n=int(input())
a=list(map(int,input().split()))
c0=[0]; c1=[0]
for _ in a:
    if _==1:
        c1+=[c1[-1]+1]; c0+=[c0[-1]]
    elif _==0:
        c1+=[c1[-1]]; c0+=[c0[-1]+1]
q=int(input())
for i in range(q):
    l,r=map(int,input().split())
    dc1=c1[r]-c1[l-1]; dc0=c0[r]-c0[l-1]
    print(["win","lose","draw"][(dc1<dc0)+(dc1==dc0)*2])

File: Tessoku_of_Competitive_Programming/B06_Lottery.nim
include atcoder/header
let N=nextInt()
let A=(@[0]&newSeqWith(N,nextInt())).cumsummed
let Q=nextInt()
for _ in 1..Q:
  let L,R=nextInt()
  let a=(A[R]-A[L-1])*2-(R-L+1)
  echo if a>0: "win" elif a<0: "lose" else: "draw"